packages/forklift-console-plugin/src/modules/Providers/hooks/useProvidersInventoryList.ts

```diff
@@ -36,27 +38,42 @@ interface UseInventoryResult {
  * It fetches data on mount and then at the specified interval.
  *
  * @param {UseInventoryParams} params - Configuration parameters for the hook.
+ * @param {V1beta1Provider[]} currNamespaceProviders - list of namespace's providers to fetch inventory data for.
  * @param {number} [params.interval=10000] - Interval (in milliseconds) to fetch new data at.
  *
  * @returns {UseInventoryResult} result - Contains the inventory data, the loading state, and the error state.
  */
 export const useProvidersInventoryList = ({
+  currNamespaceProviders = null,
   interval = 20000,
 }: UseInventoryParams): UseInventoryResult => {
   const [inventory, setInventory] = useState<ProvidersInventoryList | null>(null);
   const [loading, setLoading] = useState<boolean>(true);
   const [error, setError] = useState<Error | null>(null);
   const oldDataRef = useRef(null);
   const oldErrorRef = useRef(null);
+  const canList: boolean = useFlag('CAN_LIST_NS');
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const newInventory: ProvidersInventoryList = await consoleFetchJSON(
-          getInventoryApiUrl(`providers?detail=1`),
-        );
+        // Fetch all providers
+        if (canList) {
+          const newInventory: ProvidersInventoryList = await consoleFetchJSON(
+            getInventoryApiUrl(`providers?detail=1`),
+          );
 
-        updateInventoryIfChanged(newInventory, DEFAULT_FIELDS_TO_COMPARE);
+          updateInventoryIfChanged(newInventory, DEFAULT_FIELDS_TO_COMPARE);
+        }
+        // Fetch current namespace's providers
+        else {
+          currNamespaceProviders?.forEach(async (provider) => {
+            const newProviderInventory = await consoleFetchJSON(
+              getInventoryApiUrl(`providers/${provider.spec.type}/${provider.metadata.uid}`),
+            );
+            updateInventoryProviderIfChanged(newProviderInventory, DEFAULT_FIELDS_TO_COMPARE);
+          });
```

Comment:
can you refactor this code to look similar to the "all namespace" case, e.g.

``` ts
          const newInventory: ProvidersInventoryList = await someMethodName(namespace, providers);

          updateInventoryIfChanged(newInventory, DEFAULT_FIELDS_TO_COMPARE);
```

Reply:
I did use one method for both since it's more convenient and simple.